packages/editor-ui/src/components/TagsManager/TagsManager.vue

```diff
@@ -1,167 +1,153 @@
-				cb(newTag);
-			} catch (error) {
-				const escapedName = escape(name);
-				this.showError(
-					error,
-					this.$locale.baseText('tagsManager.showError.onCreate.title'),
-					this.$locale.baseText('tagsManager.showError.onCreate.message', {
-						interpolate: { escapedName },
-					}) + ':',
-				);
-				cb(null, error);
-			}
-		},
-
-		async onUpdate(id: string, name: string, cb: (tag: boolean, error?: Error) => void) {
-			const tag = this.tagsStore.tagsById[id];
-			const oldName = tag.name;
-
-			try {
-				if (!name) {
-					throw new Error(this.$locale.baseText('tagsManager.tagNameCannotBeEmpty'));
-				}
-
-				if (name === oldName) {
-					cb(true);
-					return;
-				}
-
-				const updatedTag = await this.tagsStore.rename({ id, name });
-				cb(!!updatedTag);
-
-				this.showMessage({
-					title: this.$locale.baseText('tagsManager.showMessage.onUpdate.title'),
-					type: 'success',
-				});
-			} catch (error) {
-				const escapedName = escape(oldName);
-				this.showError(
-					error,
-					this.$locale.baseText('tagsManager.showError.onUpdate.title'),
-					this.$locale.baseText('tagsManager.showError.onUpdate.message', {
-						interpolate: { escapedName },
-					}) + ':',
-				);
-				cb(false, error);
-			}
-		},
-
-		async onDelete(id: string, cb: (deleted: boolean, error?: Error) => void) {
-			const tag = this.tagsStore.tagsById[id];
-			const name = tag.name;
-
-			try {
-				const deleted = await this.tagsStore.deleteTagById(id);
-				if (!deleted) {
-					throw new Error(this.$locale.baseText('tagsManager.couldNotDeleteTag'));
-				}
-
-				this.tagIds = this.tagIds.filter((tagId: string) => tagId !== id);
-
-				cb(deleted);
-
-				this.showMessage({
-					title: this.$locale.baseText('tagsManager.showMessage.onDelete.title'),
-					type: 'success',
-				});
-			} catch (error) {
-				const escapedName = escape(name);
-				this.showError(
-					error,
-					this.$locale.baseText('tagsManager.showError.onDelete.title'),
-					this.$locale.baseText('tagsManager.showError.onDelete.message', {
-						interpolate: { escapedName },
-					}) + ':',
-				);
-				cb(false, error);
-			}
-		},
-
-		onEnter() {
-			if (this.isLoading) {
-				return;
-			} else if (!this.hasTags) {
-				this.onEnableCreate();
-			} else {
-				this.modalBus.emit('close');
-			}
-		},
-	},
+const tagIds = ref(props.tags.map((tag) => tag.id));
+const isCreating = ref(false);
+const modalBus = createEventBus();
+
+const tags = computed(() =>
+	tagIds.value
+		.map((tagId) => props.tags.find((tag) => tag.id === tagId))
+		.filter((tag): tag is ITag => Boolean(tag)),
+);
+const hasTags = computed(() => tags.value.length > 0);
+
+const i18n = useI18n();
+
+onMounted(() => {
+	void props.onFetchTags();
 });
+
+function onEnableCreate() {
+	isCreating.value = true;
+}
+
+function onDisableCreate() {
+	isCreating.value = false;
+}
+
+async function onCreate(name: string, cb: (tag: ITag | null, error?: Error) => void) {
```

Comment:
Some eslint warning regarding the `cb` parameter: `Identifier 'cb' is restricted.eslint[id-denylist](https://eslint.org/docs/latest/rules/id-denylist)`